### DISCUSSION TITLE
Ban the ’ character

### DISCUSSION THREAD

**Discussion by jpike88** [rome/tools#4728] [Suggestions]
It's a sneaky character and I think has no place in the codebase at all.

’ vs ' just creates confusion

Thoughts?

**Comment by Conaclos** [comment]
I don't understand your request. Do you mean in comments?

**Reply by jpike88** [reply]
in all strings

**Comment by ematipico** [comment]
@jpike88 could you add some more context to your proposal with some examples?

**Comment by Boshen** [comment]
`’` is the unicode right single quotation mark.
`'` is the ascii apostrophe.

The closest eslint rule would either be no-restricted-syntax with matching the char in strings, or this rejected proposal `https://github.com/eslint/eslint/issues/4745`.

**Reply by Conaclos** [reply]
This looks very specific...

This seems more a case of Unicode string normalization using compatibility equivalence.

I could see a rule that proposes the normalized form of a Unicode string if the string is not normalized? I could advocate for the [Normalization Form Compatibility Composition (NFKC)](https://en.wikipedia.org/wiki/Unicode_equivalence#Normal_forms).

The rule could be named `useNormalizedStrings`
